Xamarin.Forms.Platform.WPF/Renderers/FrameRenderer.cs

```diff
@@ -68,15 +68,39 @@ void UpdateBorder()
 			}
 			else
 			{
-				Control.UpdateDependencyColor(Border.BorderBrushProperty, new Color(0, 0, 0, 0));
+				Control.UpdateDependencyColor(Border.BorderBrushProperty, Color.Transparent);
 				Control.BorderThickness = new System.Windows.Thickness(0);
 			}
+
+			if (Element.HasShadow)
```

Comment:
What happens if I switch (back) to no shadow at runtime? This piece of code gets skipped, but nothing is reset, so I still keep my background?

Reply:
I changed a few things:
* Added a reset of the effect
* Seperated border from shadow update logic, I dont see how it is connected and why it should get called together
* Made the new `UpdateShadow` method virtual since the `ViewRenderer` methods are all virtual and I think thats really usefull when overriding the view to create a custom shadow
* Moved the logic to enforce a background color when the shadow should be visible to an override of the `UpdateBackground`method to ensure that it is always applied